packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -76,7 +82,15 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
 
         await this.paypalCommerceIntegrationService.loadPayPalSdk(methodId);
 
-        this.renderButton(methodId, paypalcommerce);
+        if (paypalcommerce.onInit && typeof paypalcommerce.onInit === 'function') {
+            paypalcommerce.onInit(() => this.renderButton(methodId, paypalcommerce));
+
+            return;
```

Comment:
It is not necessary to make a return here. Some custom checkouts may use this method for analytic related things and does not use it to render a button as you think. So we should not prevent wallet button rendering for cases when checkout developer pass onInit callback method to the strategy

Reply:
Thanks for the clear description